test/framework/executor_itf/executor_itf.hpp

```diff
@@ -0,0 +1,323 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TEST_FRAMEWORK_EXECUTOR_ITF_HPP
+#define IROHA_TEST_FRAMEWORK_EXECUTOR_ITF_HPP
+
+#include <memory>
+
+#include <boost/optional.hpp>
+#include "ametsuchi/command_executor.hpp"
+#include "ametsuchi/query_executor.hpp"
+#include "ametsuchi/tx_executor.hpp"
+#include "common/result.hpp"
+#include "framework/common_constants.hpp"
+#include "framework/executor_itf/executor_itf_helper.hpp"
+#include "interfaces/commands/command.hpp"
+#include "interfaces/queries/query.hpp"
+#include "logger/logger_fwd.hpp"
+#include "logger/logger_manager_fwd.hpp"
+#include "module/shared_model/command_mocks.hpp"
+#include "module/shared_model/query_mocks.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    class PublicKey;
+  }
+  namespace interface {
+    class MockCommandFactory;
+    class MockQueryFactory;
+    class Transaction;
+  }
+}  // namespace shared_model
+
+namespace iroha {
+  namespace integration_framework {
+
+    class ExecutorItf {
+     public:
+      using RunResult = iroha::expected::Result<void, std::string>;
+
+      ExecutorItf();
+
+      ~ExecutorItf();
+
+      // -------------------------- pre-init setup -----------------------------
+
+      /// Set the database connection string.
+      void setPgConnectionString(std::string pg_connection_string);
+
+      // ------------------------------- init ----------------------------------
+
+      /**
+       * Initialize Executor ITF:
+       * - connect to database 
+       * - prepare database schema
+       * - initialize command & query executors
+       * - create admin account, role, domain with all permissions
+       */
+      RunResult init();
```

Comment:
Decided offline to make a `make_unique`-style factory for construction and remove `init`.